patches/server/0988-fix-MC-65198-ITEM_CRAFTED-statistic-isn-t-always-inc.patch

```diff
@@ -0,0 +1,56 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: elmital <54907162+elmital@users.noreply.github.com>
+Date: Fri, 14 Jul 2023 18:17:38 +0200
+Subject: [PATCH] fix: MC-65198 ITEM_CRAFTED statistic isn't always incremented
+ correctly
+
+
+diff --git a/src/main/java/net/minecraft/world/inventory/ItemCombinerMenu.java b/src/main/java/net/minecraft/world/inventory/ItemCombinerMenu.java
+index ff770b9ce68a62418de0c7ed389650626fa1dcb2..e939bbc42216d688be655c47fff362906ed5f3f4 100644
+--- a/src/main/java/net/minecraft/world/inventory/ItemCombinerMenu.java
++++ b/src/main/java/net/minecraft/world/inventory/ItemCombinerMenu.java
+@@ -73,6 +73,13 @@ public abstract class ItemCombinerMenu extends AbstractContainerMenu {
+             public void onTake(Player player, ItemStack stack) {
+                 ItemCombinerMenu.this.onTake(player, stack);
+             }
++
++            // Paper start - fix MC-65198 shift clicking in result slot of a smiting table
++            @Override
++            protected void onQuickCraft(ItemStack stack, int amount) {
++                stack.onCraftedBy(player.level(), player, amount);
++            }
++            // Paper end
```

Comment:
Does this mean it runs for items from Anvil's as well? AnvilMenu extends ItemCombinerMenu and uses the same logic for setting up slots. I don't think stats should be counted when an item is taken from an anvil.

Reply:
You right I was pretty sure I've checked to avoid that but re-testing today with my changes the statistic is increased. I don't know how you prefer I change that. I could maybe simply add a check like this :
``` java 
@Override
protected void onQuickCraft(ItemStack stack, int amount) {
    if (ItemCombinerMenu.this instanceof SimthingMenu) stack.onCraftedBy(player.level(), player, amount);
}
```

Or maybe add a `onQuickCraft` method in `ItemCombinerMenu` and override it in SmithingMenu?

Reply:
I think just doing the instanceof check in the existing method is fine.